### ISSUE TITLE
nhentai tag commands

### ISSUE THREAD

**Issue by AzureAragami** [Bionus/imgbrd-grabber#2457]
A clear and concise description of what the problem is. Ex. I'm always frustrated when I can't automatically organize nhentai downloads by doujin name, artist and series.

**Describe the solution you'd like**
Similar to how danbooru grabs the tags and you can use commands such as %artist% %character%, etc... that you can do the same in nhentai.

For you to be able to use %artist%, %character%, etc commands on nhentai.

**Describe alternatives you've considered**
I have no alternatives since no command works to organize this website.

![image](https://user-images.githubusercontent.com/49356527/131227938-f29b8e41-49f5-4b76-ad14-b1110a9e890d.png)

**Comment by SvyraCosmix** [comment]
Doujin sites need the inclusion of the gallery.name token

%character%/%artist%/%gallery.name%/%filename%.%ext%

**Comment by SvyraCosmix** [comment]
List of tokens https://bionus.github.io/imgbrd-grabber/docs/filename.html

**Comment by AzureAragami** [comment]
> Doujin sites need the inclusion of the gallery.name token
> 
> %character%/%artist%/%gallery.name%/%filename%.%ext%

Thank you for the gallery.name info I couldnt figure that one out!, unfortunately the %character% and %artist% ones do not work for e, they give e an anonymous and unknown. Is there a way to get those?

**Comment by SvyraCosmix** [comment]
If you tried it on multiple galleries and it didnt work even though it does display the character and artists tags on the site then I dont know

**Comment by AzureAragami** [comment]
> If you tried it on multiple galleries and it didnt work even though it does display the character and artists tags on the site then I dont know

That is why i am requesting this feature, no doujin on nhentai gives me any tags whatsoever, I think somehow these are not loaded at all like they would be from danbooru.

**Comment by Bionus** [comment]
Indeed, currently for NHentai only the tags are available, but not their types.
I'll see what can be done.

Note that this will only apply to galleries with properly typed tags, not those who only put the artist name in the title. By "properly typed tags", I mean this:
![image](https://user-images.githubusercontent.com/882719/131871732-5e7288e9-bf9f-41cc-8bf5-c5eb76399b96.png)

There will be a mapping:
* parody => copyright
* character => character
* tag => general
* artist => artist
* groups => artist
* language => meta
* category => general

**Comment by AzureAragami** [comment]
thank you very much that would be great <3
